.github/workflows/lint-and-build.yaml

```diff
@@ -0,0 +1,30 @@
+name: lint
+
+on: [push, pull_request]
```

Comment:
same discussion we had with `omniledger`: do we need the `push` here? Supposing the `main` branch is protected from direct pushes, and all changes need to go through a PR?

Reply:
Ho, right: protections updated and `push` removed.